AzureFirewall/AzureFirewallApplicationRule.kql

```diff
@@ -0,0 +1,38 @@
+let applicationRuleAllowsHttp = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Url: " url ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule
+| where action == "Allow"
+| project-away msg_s;
+let applicationRuleAllowsHttps = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule
+| where action == "Allow" and port_destination !hasprefix "80" and rule !has "Web Category"
+| project-away msg_s;
+let applicationRuleAllowsHttpsWebCategory = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule ". Web Category: " webcategory
+| where action == "Allow"
+| project-away msg_s;
+let applicationRuleDeniesHttp = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Url: " url ". Action: " action ". " rule ". " *
+| where action == "Deny" and port_destination == 80
+| project-away msg_s;
+let applicationRuleDeniesHttps = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Action: " action ". " rule ". " *
+| where action == "Deny" and port_destination !hasprefix "80"
+| project-away msg_s;
+let applicationRuleDenies = AzureDiagnostics

```

Comment:
This branch is a good example of why the added events fields that ASIM requires such as EventType and EventResult are very important to understand the parsers output. The branch provides minimal information but, potentially, not enough to understand what it is about.

Reply:
I think I used proper column names now matching with ASIM standards....